src/wfs/src/main/java/org/geoserver/wfs/WFSInfoImpl.java

```diff
@@ -25,6 +27,8 @@ public class WFSInfoImpl extends ServiceInfoImpl implements WFSInfo {
     protected List<String> srs = new ArrayList<>();
     protected Boolean allowGlobalQueries = true;
     protected Boolean simpleConversionEnabled = false;
+    protected boolean getFeatureOutputTypeCheckingEnabled = false;
+    protected Set<String> getFeatureOutputTypes = new HashSet<>();
```

Comment:
The set might be null when XStream loads the file from disk (it's skipping normal bean initialization, including these initializers). See WFSXStreamLoader and how it's setting up the SRS list, do the same.
Check that you can start with a data dir that has no chosen formats (e..g, release data directory) and that you can still edit from GUI or REST.

Reply:
done